zkvm/src/types.rs

```diff
@@ -230,7 +230,7 @@ impl Data {
     pub fn to_predicate(self) -> Result<Predicate, VMError> {
         match self {
             Data::Opaque(data) => {
-                let point = Subslice::new(&data).read_point()?;
+                let point = SliceReader::parse(&data, |mut r| r.read_point())?;
```

Comment:
per above comment w.r.t passing &mut reader, this could become

```suggestion
                let point = SliceReader::parse(&data, |r| r.read_point())?;
```